src/NumericalAlgorithms/Interpolation/CleanUpInterpolator.hpp

```diff
@@ -63,54 +63,70 @@ struct CleanUpInterpolator {
   static void apply(db::DataBox<DbTags>& box,  // HorizonManager's box
                     const Parallel::GlobalCache<Metavariables>& /*cache*/,
                     const ArrayIndex& /*array_index*/,
-                    const double time) noexcept {
+                    const typename InterpolationTargetTag::temporal_id::type&
+                        temporal_id) noexcept {
     // Signal that this InterpolationTarget is done at this time.
     db::mutate<Tags::InterpolatedVarsHolders<Metavariables>>(
         make_not_null(&box),
-        [&time](const gsl::not_null<
-                typename Tags::InterpolatedVarsHolders<Metavariables>::type*>
-                    holders) noexcept {
+        [&temporal_id](
+            const gsl::not_null<
+            typename Tags::InterpolatedVarsHolders<Metavariables>::type*>
+                holders) noexcept {
           get<Vars::HolderTag<InterpolationTargetTag, Metavariables>>(*holders)
-              .times_when_data_has_been_interpolated.insert(time);
+              .temporal_ids_when_data_has_been_interpolated.insert(temporal_id);
         });
 
     // If we don't need any of the volume data anymore for this
-    // time, we will remove them.
-    bool this_time_is_done = true;
+    // temporal_id, we will remove them.
+    bool this_temporal_id_is_done = true;
     const auto& holders =
         db::get<Tags::InterpolatedVarsHolders<Metavariables>>(box);
     tmpl::for_each<typename Metavariables::interpolation_target_tags>(
         [&](auto tag) noexcept {
           using Tag = typename decltype(tag)::type;
-          const auto& found = get<Vars::HolderTag<Tag, Metavariables>>(holders)
-                                  .times_when_data_has_been_interpolated;
-          if (found.count(time) == 0) {
-            this_time_is_done = false;
+          if constexpr (std::is_same_v<
```

Comment:
what is happening when this is false? It looks like it will make it more likely that data is erased... (i.e. should there be an else branch setting `this_time_is_done` to false?)

Reply:
After this change, we actually have an independent count for each temporal id type (the initialization creates a separate `VolumeVarsInfo` for each temporal_id type in the set of interpolation targets), so here we're intentionally only checking and clearing data associated with the temporal id type we're currently considering.

For instance, of we have two interpolation targets,  one that uses ::Tags::TimeStepId, and one that uses ::Tags::Time, we can clean up the saved data indexed by TimeStepId after the first one has finished without waiting for the second one to complete, because we should anticipate that it receives a separately managed set of data.

Reply:
Ah, I see what my confusion was.  Data can only be erased for the temporal_id type of the current target, and it is guaranteed that at least one of the looped over interpolation target tags shares that type because the current target must be in the list of all targets...